core/templates/services/prevent-page-unload-event.service.ts

```diff
@@ -25,35 +25,50 @@ import { WindowRef } from 'services/contextual/window-ref.service';
 })
 export class PreventPageUnloadEventService {
   private listenerActive: boolean;
+  validation: Function;
+  _preventPageUnloadEventHandlerBind = undefined;
+
   constructor(private windowRef: WindowRef) {
     this.listenerActive = false;
+    this.validation = undefined;
```

Comment:
Should this value be changed?

Reply:
Do you mean the value of this.validation? I'm not sure, I've less context., can you please explain why it's needed?

Reply:
`this.validation` is actually a callback function. This function is used to trigger the "reload Site?" modal.
This function is monitored inside the addeventlistener. If true it runs the `e.returnValue = ''`, which is used to trigger the "reload Site?" modal.

Reply:
Do we need this functionality for the changes in this PR? if not, please remove unwanted changes! :)

Reply:
I originally did implement it for this PR. However, the functionality is still useful for another PR. So, do I remove it?

Reply:
I removed the changes. Will include it in the PR it is meant for.